File: Excel/516 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/516 Product of Digits of Result is Equal to Number.xlsx"
input = read_excel(path, range = "A1:A10")
test  = read_excel(path, range = "B1:B10") %>% 
  mutate(`Answer Expected` = as.character(`Answer Expected`))

find_smallest_number_with_digit_product = function(n) {
  if (n == 0) return(10)
  if (n == 1) return(1)
  factors = c()
  for (i in 9:2) {
    while (n %% i == 0) {
      factors = c(factors, i)
      n = n / i
    }
  }
  if (n > 1) return("NP")
  return(paste(sort(factors), collapse = ""))
}

result = input %>%
  mutate(`Answer Expected` = map_chr(Number, find_smallest_number_with_digit_product)) %>%
  select(2)

identical(result, test)
# [1] TRUE

File: Excel/500-599/516/516 Challenge.py
import pandas as pd

path = "516 Product of Digits of Result is Equal to Number.xlsx"
input = pd.read_excel(path, usecols="A")
test  = pd.read_excel(path, usecols="B")

def find_smallest_number_with_digit_product(n):
    if n == 0:
        return 10
    if n == 1:
        return 1
    
    factors = []
    
    for i in range(9, 1, -1):
        while n % i == 0:
            factors.append(i)
            n //= i
    
    return int(''.join(map(str, sorted(factors)))) if n == 1 else "NP"

result = [find_smallest_number_with_digit_product(n) for n in input["Number"]]
result_df = pd.DataFrame(result, columns=["Answer Expected"])
print(result_df.equals(test))   # True